src/Processors/Transforms/JoiningTransform.h

```diff
@@ -147,18 +152,31 @@ class DelayedJoinedBlocksTransform : public IProcessor
 class DelayedJoinedBlocksWorkerTransform : public IProcessor
 {
 public:
-    explicit DelayedJoinedBlocksWorkerTransform(Block output_header);
+    explicit DelayedJoinedBlocksWorkerTransform(
+        Block left_header_,
+        Block output_header_,
+        size_t max_block_size_,
+        JoinPtr join_);
 
     String getName() const override { return "DelayedJoinedBlocksWorkerTransform"; }
 
     Status prepare() override;
     void work() override;
 
 private:
+    Block left_header;
```

Comment:
It'd be nice to restructure the variable's definitions with comments to make it obvious for which stage which variables are used (see stages mentioned below).